worlds/shivers/__init__.py

```diff
@@ -56,162 +63,192 @@ def create_regions(self) -> None:
-        library_region = self.multiworld.get_region("Library", self.player)
-        librarylocation = self.random.choice([loc for loc in library_region.locations if not loc.name.startswith("Accessible:")])
+        # Place library escape items. Choose a location to place the escape item
+        library_region = self.get_region("Library")
+        library_location = self.random.choice(
+            [loc for loc in library_region.locations if not loc.name.startswith("Storage: ")]
+        )
 
-        #Roll for which escape items will be placed in the Library
+        # Roll for which escape items will be placed in the Library
         library_random = self.random.randint(1, 3)
-        if library_random == 1: 
-            librarylocation.place_locked_item(self.create_item("Crawling"))
+        if library_random == 1:
+            library_location.place_locked_item(self.create_item("Crawling"))
+            item_pool = [item for item in item_pool if item.name != "Crawling"]
+        elif library_random == 2:
+            library_location.place_locked_item(self.create_item("Key for Library"))
+            item_pool = [item for item in item_pool if item.name != "Key for Library"]
+        elif library_random == 3:
+            library_location.place_locked_item(self.create_item("Key for Three Floor Elevator"))
+            library_location_2 = self.random.choice(
+                [loc for loc in library_region.locations if
+                 not loc.name.startswith("Storage: ") and loc != library_location]
+            )
+            library_location_2.place_locked_item(self.create_item("Key for Egypt Room"))
+            item_pool = [item for item in item_pool if
+                         item.name not in ["Key for Three Floor Elevator", "Key for Egypt Room"]]
+
+        # If front door option is on, determine which set of keys will
+        # be used for lobby access and add front door key to item pool
+        lobby_access_keys = 0
+        if self.options.front_door_usable:
+            lobby_access_keys = self.random.randint(0, 1)
+            item_pool.append(self.create_item("Key for Front Door"))
+        else:
+            item_pool.append(self.create_item("Heal"))
 
-            itempool = [item for item in itempool if item.name != "Crawling"]
+        def set_lobby_access_keys(items: Dict[str, int]):
+            if lobby_access_keys == 0:
+                items["Key for Underground Lake"] = 1
+                items["Key for Office Elevator"] = 1
+                items["Key for Office"] = 1
+            else:
+                items["Key for Front Door"] = 1
 
-        elif library_random == 2: 
-            librarylocation.place_locked_item(self.create_item("Key for Library Room"))
+        # Lobby access:
+        if self.options.lobby_access == 1:
+            set_lobby_access_keys(self.multiworld.early_items[self.player])
+        elif self.options.lobby_access == 2:
```

Comment:
It might be good to update the documentation of the option here as well, since it says "Keys are placed locally" but this is placing them locally *and* early

Reply:
I was confused by what you meant, but I see, yes that is local_early_items which is different from just local_items. Wonder if I should change the option name then. (Ironically using local or local_early realistically wouldn't make much of a difference here except that the office elevator key and office keys could be placed very slightly later, I should discuss this with GodlFire)